server/handlers/genomic_annotations.go

```diff
@@ -1 +1,155 @@
 package handlers
+
+import (
+	"errors"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/ldsec/medco-connector/restapi/models"
+	"github.com/ldsec/medco-connector/restapi/server/operations/genomic_annotations"
+	utilserver "github.com/ldsec/medco-connector/util/server"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+var genomicAnnotationsTypes = []string{"variant_name", "protein_change", "hugo_gene_symbol"}
+
+func MedCoGenomicAnnotationsGetValuesHandler(params genomic_annotations.GetValuesParams, principal *models.User) middleware.Responder {
+
+	err := utilserver.DBConnection.Ping()
+	if err != nil {
+		logrus.Error("Impossible to connect to DB: " + err.Error())
+		return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+			Message: "Impossible to connect to DB " + err.Error(),
+		})
+	}
+
+	var query string
+
+	query, err = buildGetValuesQuery(params)
+	if err != nil {
+		logrus.Error("Query execution error " + err.Error())
+		return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+			Message: "Query execution error: " + err.Error(),
+		})
+	}
+
+	//escaping * characters
+	rows, err := utilserver.DBConnection.Query(query, strings.ReplaceAll(params.Value, "*", "\\*"), *params.Limit)
+	if err != nil {
+		logrus.Error("Query execution error: " + err.Error())
+		return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+			Message: "Query execution error: " + err.Error(),
+		})
+	}
+	defer rows.Close()
+
+	var annotations []string
+	var annotation string
+
+	for rows.Next() {
+		err := rows.Scan(&annotation)
+		if err != nil {
+			logrus.Error("Query result reading error: " + err.Error())
+			return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+				Message: "Query result reading error: " + err.Error(),
+			})
+		}
+		annotations = append(annotations, annotation)
+	}
+
+	if len(annotations) > 0 {
+		return genomic_annotations.NewGetValuesOK().WithPayload(annotations)
+	} else {
+		return genomic_annotations.NewGetValuesNotFound()
```

Comment:
Here the behavior should be different: not finding any corresponding value is actually a valid answer (200 and not 404) and should return just an empty array.

The 404 returned with `NewGetValuesNotFound` should actually be used when the annotation name is not found (check out the swagger file it is specific there: the `NewGetValuesNotFound` object exists because the 404 response is specified there).